app/src/main/java/net/squanchy/navigation/RoutingActivity.java

```diff
@@ -24,30 +29,60 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         RoutingComponent component = RoutingInjector.obtain(this);
         deepLinkRouter = component.deepLinkRouter();
         navigator = component.navigator();
-
-        SignInService signInService = component.signInService();
-        subscription = signInService.signInAnonymouslyIfNecessary().subscribe();
+        signInService = component.signInService();
+        firstStartPersister = component.firstStartPersister();
     }
 
     @Override
     protected void onStart() {
         super.onStart();
 
+        subscription = signInService.signInAnonymouslyIfNecessary()
+                .subscribe(this::proceedToRouting, this::handleSignInError);
+    }
+
+    private void proceedToRouting() {
         Intent intent = getIntent();
         if (deepLinkRouter.hasDeepLink(intent)) {
             String intentUriString = intent.getDataString();
             Timber.i("Deeplink detected, navigating to \"%s\"", intentUriString);
             deepLinkRouter.navigateTo(intentUriString);
         } else {
-            navigator.toSchedule();
+            navigator.toHomePage();
+        }
+
+        firstStartPersister.storeHasBeenStarted();
+        finish();
+    }
+
+    private void handleSignInError(Throwable throwable) {
+        Timber.e(throwable, "Error while signing in on routing");
+        if (isFirstStart()) {
```

Comment:
Well it does in the sense we might want to communicate something different in case, maybe non MVP